stdlib/src/collections/deque.mojo

```diff
@@ -0,0 +1,425 @@
+        """
+        return (self._tail - self._head) & (self._capacity - 1)
+
+    fn __getitem__(ref [_]self, idx: Int) -> ref [self] ElementType:
+        """Gets the deque element at the given index.
+
+        Args:
+            idx: The index of the element.
+
+        Returns:
+            A reference to the element at the given index.
+        """
+        normalized_idx = idx
+
+        debug_assert(
+            -len(self) <= normalized_idx < len(self),
+            "index: ",
+            normalized_idx,
+            " is out of bounds for `Deque` of size: ",
+            len(self),
+        )
+
+        if normalized_idx < 0:
+            normalized_idx += len(self)
+
+        offset = self._physical_index(self._head + normalized_idx)
+        return (self._data + offset)[]
+
+    # ===-------------------------------------------------------------------===#
+    # Methods
+    # ===-------------------------------------------------------------------===#
+
+    fn append(inout self, owned value: ElementType):
+        """Appends a value to the right side of the deque.
+
+        Args:
+            value: The value to append.
+        """
+        # checking for positive _maxlen first is important for speed
+        if self._maxlen > 0 and len(self) == self._maxlen:
+            (self._data + self._head).destroy_pointee()
+            self._head = self._physical_index(self._head + 1)
+
+        (self._data + self._tail).init_pointee_move(value^)
+        self._tail = self._physical_index(self._tail + 1)
+
+        if self._head == self._tail:
+            self._realloc(self._capacity << 1)
+
+    fn extend(inout self, owned values: List[ElementType]):
+        """Extends the right side of the deque by consuming elements of the list argument.
+
+        Args:
+            values: List whose elements will be added at the right side of the deque.
+        """
+        n_move_total, n_move_self, n_move_values, n_pop_self, n_pop_values = (
+            self._compute_pop_and_move_counts(len(self), len(values))
+        )
+
+        # pop excess `self` elements
+        for _ in range(n_pop_self):
+            (self._data + self._head).destroy_pointee()
+            self._head = self._physical_index(self._head + 1)
+
+        # move from `self` to new location if we have to re-allocate
+        if n_move_total >= self._capacity:
+            self._prepare_for_new_elements(n_move_total, n_move_self)
+
+        # we will consume all elements of `values`
+        values_data = values.steal_data()
+
+        # pop excess elements from `values`
+        for i in range(n_pop_values):
+            (values_data + i).destroy_pointee()
+
+        # move remaining elements from `values`
+        src = values_data + n_pop_values
+        for i in range(n_move_values):
+            (src + i).move_pointee_into(self._data + self._tail)
+            self._tail = self._physical_index(self._tail + 1)
+
+    @doc_private
```

Comment:
**Suggestion** There's no need to explicitly mark these with `@doc_private` since they are prefixed with `_` in the function identifier.  The doc-gen tooling automatically detects this case and won't generate API docs on the website for private functions.  I can remove them here and elsewhere for you, no biggie - but just FYI.

Reply:
Noted. Will remove @doc_private in the next PR.